Gems/Atom/RPI/Code/Source/RPI.Public/Material/Material.cpp

```diff
@@ -60,18 +60,19 @@ namespace AZ
             ScopedValue isInitializing(&m_isInitializing, true, false);
 
             // All of these members must be reset if the material can be reinitialized because of the shader reload notification bus
+            m_materialAsset = { &materialAsset, AZ::Data::AssetLoadBehavior::PreLoad };
             m_shaderResourceGroup = {};
             m_rhiShaderResourceGroup = {};
             m_materialProperties = {};
             m_generalShaderCollection = {};
             m_materialPipelineData = {};
-            m_materialAsset = { &materialAsset, AZ::Data::AssetLoadBehavior::PreLoad };
+
             ShaderReloadNotificationBus::MultiHandler::BusDisconnect();
             if (!m_materialAsset.IsReady())
             {
-                // We will call this function again later when the asset is ready.
-                Data::AssetBus::Handler::BusConnect(m_materialAsset.GetId());
-                return RHI::ResultCode::Success;
+                AZ_Error(s_debugTraceName, false, "Material::Init failed because acid is not ready. materialAsset uuid=%s",
```

Comment:
`acid` 😄 
```suggestion
                AZ_Error(s_debugTraceName, false, "Material::Init failed because asset is not ready. materialAsset uuid=%s",
```